### PR TITLE
Improve template

### PR DESCRIPTION
**Pull request by thoean**
* Mostly a few tweaks
* Added editorconfig
* Added eslint
* Upgraded all dependencies to latest version
* Made aws-sdk a dev dependency

I made them all individual commits. So it should be easy to select only those that should go in.

### CODE REVIEW

**Comment by wparad** [comment]
Making aws-sdk a dev dependency will prevent any client of the library from specifying the version they want.  Can we leave that change out and possibly tackle a different way further down the line?

**Comment by wparad** [comment]
Wait it should have aws-sdk as optional, but it seem in here is in the direct dependencies... stratge

**Comment by wparad** [comment]
Oh its the template package json.  Then my only feedback is that the openapi-json might be better left as "^4.3"

File: bin/template/package.json [hunk 1]
```diff
@@ -8,28 +8,36 @@
...
-    "jwk-to-pem": "^1.2.6",
-    "microservice-utilities": "^0.1",
-    "openapi-factory": "^4.3"
+    "axios": "^0.18.0",
+    "ci-build-tools": "^1.0.13",
+    "fs-extra": "^7.0.1",
+    "jsonwebtoken": "^8.4.0",
+    "jwk-to-pem": "^2.0.0",
+    "microservice-utilities": "^0.1.78",
+    "openapi-factory": "^4.3.178"
```
Comment (wparad) [inline bin/template/package.json]: If we could leave this as `^4.3` unless there was a specific reason to change.  My goal was for new users to automatically be able to get the latest version without having to constantly come back here to update it.

File: bin/template/package.json [hunk 2]
```diff
@@ -8,28 +8,36 @@
...
   },
   "devDependencies": {
-    "aws-architect": "^6.1",
-    "chai": "^3.2.0",
-    "commander": "^2.5.0",
+    "aws-architect": "^6.1.489",
```
Comment (wparad) [inline bin/template/package.json]: Same with this one, unless there was something in particular that was a concern.  I'd like to understand this change.

**Comment by wparad** [comment]
one small thing on the library versions in the package.json, other than that.  It's good and I will merge after that.

**Comment by thoean** [comment]
I've made the changes for `openapi-factory` and `aws-architect` to automatically get the next minor version. I wonder whether that wouldn't be a good practive for all of the libraries for the template.

**Pull request merged**